src/test/compile-fail/borrowck/borrowck-borrow-overloaded-auto-deref-mut.rs

```diff
@@ -99,7 +99,7 @@ fn assign_field1<'a>(x: Own<Point>) {
 }
 
 fn assign_field2<'a>(x: &'a Own<Point>) {
-    x.y = 3; //~ ERROR cannot assign
+    x.y = 3; //~ ERROR cannot borrow
```

Comment:
Is this change a bug-fix?
It seems to typeck as `(*DerefMut::deref_mut(&mut *x)).y = 3;` where previously it was `(*Deref::deref(&*x)).y = 3;`.